feelpp/feel/feeldiscr/createsubmesh.hpp

```diff
@@ -393,7 +400,8 @@ CreateSubmeshTool<MeshType,IteratorRange,TheTag>::build( mpl::int_<MESH_ELEMENTS
             } // for (unsigned int n=0 ... )
 
             // Add an equivalent element type to the new_mesh
-            auto const& e = newMesh->addElement( newElem,true );
+            auto eit = newMesh->addElement( newElem,true );
+            auto const& e = eit.first->second;
```

Comment:
and here extract explicit the different members of the pair so that it is more explicit?
that way the code is more readable.

those two lines are everywhere and I(and probably others) need to remember what is the content of the pair/tuple.
may be we could query/replace the code automatically ?

Reply:
@prudhomm today, I have started the changes that you propose with addElement. an automatically way is quite dangerous.

Reply:
ok. so by hand?

Reply:
Yes, there are not so many. I compile currently the testsuite